lib/lambda_ethereum_consensus/p2p/incoming_request_handler.ex

```diff
@@ -117,8 +119,46 @@ defmodule LambdaEthereumConsensus.P2P.IncomingRequestHandler do
     end
   end
 
-  def handle_req("beacon_blocks_by_range/2/ssz_snappy", _message_id, _message),
-    do: :not_implemented
+  def handle_req("beacon_blocks_by_range/2/ssz_snappy", message_id, message) do
+    with <<24, snappy_blocks_by_range_request::binary>> <- message,
+         {:ok, ssz_blocks_by_range_request} <- Snappy.decompress(snappy_blocks_by_range_request),
+         {:ok, blocks_by_range_request} <-
+           Ssz.from_ssz(ssz_blocks_by_range_request, SszTypes.BeaconBlocksByRangeRequest) do
+      blocks_by_range_request
+      |> inspect(limit: :infinity)
+      |> then(&"[Received BlocksByRange Request] '#{&1}'")
+      |> Logger.debug()
+
+      ## TODO: there should be check that the `start_slot` is not older than the `oldest_slot_with_block`
+      %SszTypes.BeaconBlocksByRangeRequest{start_slot: start_slot, count: count} =
+        blocks_by_range_request
+
+      slot_coverage = start_slot + count
+
+      blocks =
+        start_slot..slot_coverage
+        |> Enum.reduce_while([], fn slot, current_blocks ->
+          blocks = case StateStore.get_state_by_slot(slot) do
+            {:ok, state} -> current_blocks ++ BlockStore.get_blocks(state.block_roots)
+            {:error, _} -> current_blocks
+            :not_found -> current_blocks
+          end
+          if length(blocks) <= ChainSpec.get("MAX_REQUEST_BLOCKS") do
+            {:cont, blocks}
+          else
+            {:halt, blocks}
+          end
+        end)
+
+      blocks_by_range_response = %SszTypes.BeaconBlocksByRangeResponse{body: blocks}
```

Comment:
This wouldn't work, sadly. The spec shows the response as an SSZ container, but that isn't actually correct. The response is given as a stream of chunks, each corresponding to a block, and prefixed with a header. An example would be:

```
request: 3 blocks: A, B (we don't have B)
response: `0 result (OK)` <> `ForkDigest-context` <> `size varint` <> `A (ssz_snappy encoded)` <>
           `3 result (not found)` <> `error_message`
```
Some relevant links:
- https://github.com/lambdaclass/lambda_ethereum_consensus/blob/d3344795df9d55c52c84da84ef12da8695c57243/docs/specs/p2p-interface.md?plain=1#L770-L771
- https://github.com/lambdaclass/lambda_ethereum_consensus/blob/main/docs/specs/p2p-interface.md?plain=1#L695

Reply:
result in this case is like a status_code, right?